vowpalwabbit/parse_example_json.h

```diff
@@ -1038,19 +1028,19 @@ struct Context
     if (ns.feature_count > 0)
     {
       auto feature_group = ns.feature_group;
-      // avoid duplicate insertion
-      for (unsigned char ns_char : ex->indices)
-        if (ns_char == feature_group)
-          goto done;
-
-      ex->indices.push_back(feature_group);
+      // Do not insert feature_group if it already exists.
+      if(std::find(ex->indices.begin(), ex->indices.end(), feature_group) == ex->indices.end())
+      {
+        ex->indices.push_back(feature_group);
+      }
     }
 
-  done:
-    return namespace_path.pop().return_state;
+    auto return_state = namespace_path.back().return_state;
```

Comment:
[minor] namespace_path.back()  ->   ns?

Reply:
I'd like to leave it as is because it is the normal way you expect to see the back of a queue being popped